src/server/stream_family.cc

```diff
@@ -2948,21 +2948,33 @@ void XReadGeneric2(CmdArgList args, ConnectionContext* cntx, bool read_group) {
 
   auto* tx = cntx->transaction;
 
-  // Determine if streams have entries
+  // Determine if streams have entries or any error occured
   AggregateValue<optional<facade::ErrorReply>> err;
   atomic_bool have_entries = false;
```

Comment:
why atomic_bool? hops ensure sequential memory order in the coordinator so seems can be done with a regular bool?

Reply:
Concurrently writes? 🙂